utils/data_utils.py

```diff
@@ -1,8 +1,66 @@
 import os
+import requests
+from urllib.parse import urlparse
 
 
-def get_data_options(dir="data"):
-    return [file for file in os.listdir(f"{dir}/") if file.endswith("tiff")]
+def get_data_project_names(dir="data"):
```

Comment:
Just need to add the following otherwise this will throw an error if the `data` dir doesn't exist: 

```
    if not os.path.exists(dir):
        os.mkdir(dir) 
```

Reply:
I don't think this is actually the problem we're hitting though - the logs say it can't find "data/None" and I think it's because of this chain of events:
- DATA_OPTIONS = data_utils.get_data_project_names() is at the top of the control_bar.py file so it only runs once, on app deployment. The get_data_project_names() function works just fine, the data folder always exists because we created one in our code base to begin with. However, there are no images or project names in that data folder so we get to the next block of code
- dmc.Select(
      id="project-name-src",
      data=DATA_OPTIONS,
      value=DATA_OPTIONS[0] if DATA_OPTIONS else None,
      placeholder="Select an image to view...",
  ),
- Which means that the project-name-src value is None and then the update_slider_values() callback in image_viewer.py tries to run the data_utils.get_tiff_files(project_name) line but the project_name is None leading to the log error we see of FileNotFoundError: [Errno 2] No such file or directory: 'data/None'

Reply:
So the problem is that the DEV_download_google_sample_data() function getting run in app.py is not guaranteed to be the first thing that happens. We need to move it to a predeploy script so that on app's first load, there is already that data there.